File: sync/v1/document/create-document-1/create-document-1.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Sync.V1.Service;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var data = new Dictionary<string, Object>()
        {
            {"date_updated", "2018-02-14 12:24:31.843662"},
            {"movie_title", "On The Line"},
            {"show_times", new object [] {
                "12:30:00Z",
                "14:45:00Z",
                "15:30:00Z",
                "17:45:00Z"
            }},
            {"starring", new object [] {
                "Lance Bass",
                "Joey Fatone"
            }},
            {"genre", "Romance"}
        };

        var document = DocumentResource.Create(
            data: data,
            ttl: 1814400,
            uniqueName: "MyFirstDocument",
            pathServiceSid: "ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        );

        Console.WriteLine(document.Sid);
    }
}

File: sync/v1/document/create-document-1/create-document-1.5.x.php
<?php

// Update the path below to your autoload.php,
// see https://getcomposer.org/doc/01-basic-usage.md
require_once '/path/to/vendor/autoload.php';

use Twilio\Rest\Client;

// Find your Account Sid and Auth Token at twilio.com/console
$sid    = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
$token  = "your_auth_token";
$twilio = new Client($sid, $token);

$document = $twilio->sync->v1->services("ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
                             ->documents
                             ->create(array(
                                          "data" => array(
                                              "date_updated" => "2018-02-14 12:24:31.843662",
                                              "movie_title" => "On The Line",
                                              "show_times" => array(
                                                  "12:30:00Z",
                                                  "14:45:00Z",
                                                  "15:30:00Z",
                                                  "17:45:00Z"
                                              ),
                                              "starring" => array(
                                                  "Lance Bass",
                                                  "Joey Fatone"
                                              ),
                                              "genre" => "Romance"
                                          ),
                                          "ttl" => 1814400,
                                          "uniqueName" => "MyFirstDocument"
                                      )
                             );

print($document->sid);